src/dataflow/src/sink/kafka.rs

```diff
@@ -351,6 +387,30 @@ where
         for _ in 0..connector.fuel {
             if let Some((ts, rows)) = ready_rows.front() {
                 state = match state {
+                    SendState::Init => {
```

Comment:
Is there a compelling reason this logic _has_ to be in this loop / part of the state machine instead of closer to where we initialize the producer / outside of the `sink_logic` closure?

basically, the way I see it, I currently have to audit the code to make sure it obeys the "Init only happens once" property, but all of the code outside of the `sink_logic` code already happens only once by default and so it seems simpler to have it there?

Reply:
Since we create a sink operator for each worker, doing this logic outside of the state machine would limit us to a transactional ID that includes the worker/thread ID. Here inside the `ready_rows` read is the only place to put it such that we initialize transactions only on the operator that's responsible for publishing.

If it's any consolation, things go wrong _very_ quickly if the "init only once and on one operator" logic is off.

Reply:
Ah ok. It might be worth adding a comment that tells us this should happen only once across all workers as this was not obvious to me.